pkg/controller/quota/quota.go

```diff
@@ -0,0 +1,135 @@
+package quota
+
+import (
+	"fmt"
+
+	apiv1 "github.com/acorn-io/acorn/pkg/apis/internal.acorn.io/v1"
+	adminv1 "github.com/acorn-io/acorn/pkg/apis/internal.admin.acorn.io/v1"
+	"github.com/acorn-io/acorn/pkg/condition"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/acorn-io/acorn/pkg/config"
+	"github.com/acorn-io/baaah/pkg/router"
+)
+
+// WaitForAllocation blocks the appInstance from being deployed until quota has been allocated.
+func WaitForAllocation(req router.Request, resp router.Response) error {
+	quotaRequest := req.Object.(*adminv1.QuotaRequestInstance)
+
+	// Attempt to get the appInstance for this quota request. It should exist with the name and namespace of the
+	// QuotaRequest being processed.
+	appInstance := &apiv1.AppInstance{}
+	err := req.Client.Get(req.Ctx, router.Key(quotaRequest.Namespace, quotaRequest.Name), appInstance)
+	if err != nil {
+		return err
+	}
+
+	// Create a condition setter for AppInstanceConditionQuotaAllocated, which blocks the appInstance from being deployed
+	// until quota has been allocated.
+	status := condition.Setter(appInstance, resp, apiv1.AppInstanceConditionQuotaAllocated)
+
+	// Determine how to proceed depending on if the quotaRequest exists and what it has written to its status
+	if quotaRequest.Status.Condition(adminv1.QuotaRequestCondition).Error {
+		status.Error(fmt.Errorf("%v", quotaRequest.Status.Condition(adminv1.QuotaRequestCondition).Message))
+	} else if quotaRequest.Generation != quotaRequest.Status.ObservedGeneration {
+		status.Unknown("waiting for quota allocation")
+	} else if quotaRequest.Status.Condition(adminv1.QuotaRequestCondition).Success {
+		status.Success()
+	}
+
+	return req.Client.Status().Update(req.Ctx, appInstance)
```

Comment:
This was originally an AppInstance handler and I moved away from it because I was noticing some weird behavior in the integration test where the QuotaRequest status would be updated but this handler would never run since (I think) the AppInstance wasn't actually changed

Reply:
If the `QuotaRequest` is created by a handler on the `AppInstance`, then changes to the `QuotaRequest` should re-enqueue the corresponding `AppInstance`.

Reply:
Alright, I have a fix for this that also has the tests passing. Resolving but let me know if you have other comments.